docs/source/actor-rhel7-to-rhel8.md

```diff
@@ -0,0 +1,332 @@
+
+### snactor
+
+Separate tool provided by Leapp to help the process of creating and executing an actor.
+
+You can see _snactor_ source code [here](https://github.com/oamg/leapp/tree/master/leapp/snactor).
+
+## Creating an actor
+
+Every actor needs to be inside a so-called “Leapp repository”, otherwise it won’t be visible to Leapp. A Leapp repository groups actors and many other things which will be discussed later, like models, workflows, tags and topics. You can find all Leapp repositories under /usr/share/leapp-repository/repositories. A Leapp repository can be recognized by containing .leapp folder:
+
+```shell
+$ find -L /etc/leapp/repos.d/ -name ".leapp" -type d | xargs dirname
+/etc/leapp/repos.d/common
+/etc/leapp/repos.d/system_upgrade/el7toel8
+```
+
+First, you need to register repositories with snactor:
+
+```shell
+$ snactor repo find --path /etc/leapp/repos.d/
+Registering /etc/leapp/repos.d/system_upgrade/el7toel8
+Registering /etc/leapp/repos.d/common
+```
+
+After registering the repositories, you can move inside any of these repositories and use snactor to create a boilerplate of a new actor:
+
+```shell
+# cd /etc/leapp/repos.d/system_upgrade/el7toel8
+# snactor new-actor MyNewActor
+New actor MyNewActor has been created at /usr/share/leapp-repository/repositories/system_upgrade/el7toel8/actors/mynewactor/actor.py
+# cd /usr/share/leapp-repository/repositories/system_upgrade/el7toel8/actors/mynewactor/
+# tree
+.
+├── actor.py
+└── tests
+```
+
+The main file of the actor is the actor.py in which you’ll write code for logic of the actor.
+
+For further information about how create an actor read this [document](https://leapp.readthedocs.io/en/latest/first-actor.html)
+
+## Including an actor in the RHEL 7 to 8 upgrade process
+
+Until now, you have created boilerplate of a new actor and made it visible to Leapp. But, Leapp needs some more information about what to do with the actor. Specifically, in which **“workflow”** and in which **“phase”** the actor should be executed. A workflow is a sequence of phases. The only workflow available now is the one solving the upgrade of RHEL 7 to RHEL 8. Each phase is a set of actors that will be executed one after another before the next phase starts. To find out in which workflow and phase should the actor be executed, Leapp looks for **“tags”**. To be part of RHEL 7 to RHEL 8 upgrade workflow, an actor needs to be tagged with **IPUWorkflowTag**.
+
+The phases of the IPUWorkflow (in order) are: **Facts Collection, Checks, Report, Download, Upgrade RamDisk Preparation, Upgrade RamDisk Start, Late Tests, Preparation, RPM Upgrade, Application Upgrade, Third Party Applications, Finalization, Reboot** and **First Boot**. Each phase has a specific tag that marks an actor as being part of that phase. You can find descriptions of all the phases and their tags [here](https://github.com/oamg/leapp-repository/blob/master/repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py).
```

Comment:
I know it was in the gdoc, but Reboot is not and has not been a phase. Please remove it.